services/softlayer_product_package.go

```diff
@@ -56,3 +62,94 @@ func (slpp *softLayer_Product_Package_Service) GetItemPricesBySize(packageId int
 
 	return itemPrices, nil
 }
+
+func (slpp *softLayer_Product_Package_Service) GetItemsByType(packageType string) ([]datatypes.SoftLayer_Product_Item, error) {
+
+	productPackage, err := slpp.GetOnePackageByType(packageType)
+	if err != nil {
+		return []datatypes.SoftLayer_Product_Item{}, err
+	}
+
+	return slpp.GetItems(productPackage.Id)
+}
+
+func (slpp *softLayer_Product_Package_Service) GetItems(packageId int) ([]datatypes.SoftLayer_Product_Item, error) {
+
+	objectMasks := []string{
+		"id",
+		"capacity",
+		"description",
+		"prices.id",
+		"prices.categories.id",
+		"prices.categories.name",
+	}
+
+	response, err := slpp.client.DoRawHttpRequestWithObjectMask(fmt.Sprintf("%s/%d/getItems.json", slpp.GetName(), packageId), objectMasks, "GET", new(bytes.Buffer))
+	if err != nil {
+		return []datatypes.SoftLayer_Product_Item{}, err
+	}
+
+	productItems := []datatypes.SoftLayer_Product_Item{}
+	err = json.Unmarshal(response, &productItems)
+	if err != nil {
+		return []datatypes.SoftLayer_Product_Item{}, err
+	}
+
+	return productItems, nil
+}
+
+func (slpp *softLayer_Product_Package_Service) GetOnePackageByType(packageType string) (datatypes.Softlayer_Product_Package, error) {
+
+	productPackages, err := slpp.GetPackagesByType(packageType)
+	if err != nil {
+		return datatypes.Softlayer_Product_Package{}, err
+	}
+
+	if len(productPackages) == 0 {
+		return datatypes.Softlayer_Product_Package{}, errors.New(fmt.Sprintf("No packages available for type '%s'.", packageType))
+	}
+
+	return productPackages[0], nil
+}
+
+func (slpp *softLayer_Product_Package_Service) GetPackagesByType(packageType string) ([]datatypes.Softlayer_Product_Package, error) {
+
+	objectMasks := []string{
+		"id",
+		"name",
+		"description",
+		"isActive",
+		"type.keyName",
+	}
+
+	filterObject := string(`{"type":{"keyName":{"operation":"` + packageType + `"}}}`)
+
+	response, err := slpp.client.DoRawHttpRequestWithObjectFilterAndObjectMask(fmt.Sprintf("%s/getAllObjects.json", slpp.GetName()), objectMasks, filterObject, "GET", new(bytes.Buffer))
+	if err != nil {
+		return []datatypes.Softlayer_Product_Package{}, err
+	}
+
+	productPackages := []*datatypes.Softlayer_Product_Package{}
+	err = json.Unmarshal(response, &productPackages)
+	if err != nil {
+		return []datatypes.Softlayer_Product_Package{}, err
+	}
+
+	// Remove packages designated as OUTLET
+	nonOutletPackages := slpp.filterProducts(productPackages, func(productPackage *datatypes.Softlayer_Product_Package) bool {
```

Comment:
For this particular method we used the SoftlLayer Python client as a guide. Please see the method `get_packages_of_type` in the [SoftLayer Python client code](https://github.com/softlayer/softlayer-python/blob/master/SoftLayer/managers/ordering.py) for details. Specifically:

```
        """Get packages that match a certain type.
        Each ordering package has a type, so return all packages that match
        the types we are looking for
        :param list package_types: List of strings representing the package
                                   type keynames we are interested in.
        :param string mask: Mask to specify the properties we want to retrieve
        """
```

Please let me know if this is sufficient or if we should add more to this (perhaps a comment).